File: app/src/main/java/com/example/kaola/myapplication/designpattern/decorator/java/factory/MeatDumplings.java
package com.example.kaola.myapplication.designpattern.decorator.java.factory;

/**
 * @author zhangchao on 2019-07-18.
 */

public class MeatDumplings extends ADumplings {
    Dumplings dumplings;

    @Override
    public void makeDumplings() {
        dumplings = new Dumplings();
        dumplings.setName("我是肉饺子");
    }

    @Override
    public Dumplings getDumpling() {
        return dumplings;
    }
}

File: app/src/main/java/com/example/kaola/myapplication/designpattern/decorator/kotlin/factory/MeatDumplings.kt
package com.example.kaola.myapplication.designpattern.decorator.kotlin.factory

/**
 * @author zhangchao on 2019-07-18.
 */

class MeatDumplings : ADumplings() {
    lateinit var mDumplings: Dumplings
    override fun getDumplings(): Dumplings {
        return mDumplings
    }

    override fun makeDumplings() {
        mDumplings = Dumplings("肉馅饺子")
    }


}